src/Symfony/Component/VarDumper/Caster/XmlReaderCaster.php

```diff
@@ -0,0 +1,102 @@
+ */
+class XmlReaderCaster
+{
+    private static $nodeTypes = array(
+        \XmlReader::NONE => 'NONE',
+        \XmlReader::ELEMENT => 'ELEMENT',
+        \XmlReader::ATTRIBUTE => 'ATTRIBUTE',
+        \XmlReader::TEXT => 'TEXT',
+        \XmlReader::CDATA => 'CDATA',
+        \XmlReader::ENTITY_REF => 'ENTITY_REF',
+        \XmlReader::ENTITY => 'ENTITY',
+        \XmlReader::PI => 'PI',
+        \XmlReader::COMMENT => 'COMMENT',
+        \XmlReader::DOC => 'DOC',
+        \XmlReader::DOC_TYPE => 'DOC_TYPE',
+        \XmlReader::DOC_FRAGMENT => 'DOC_FRAGMENT',
+        \XmlReader::NOTATION => 'NOTATION',
+        \XmlReader::WHITESPACE => 'WHITESPACE',
+        \XmlReader::SIGNIFICANT_WHITESPACE => 'SIGNIFICANT_WHITESPACE',
+        \XmlReader::END_ELEMENT => 'END_ELEMENT',
+        \XmlReader::END_ENTITY => 'END_ENTITY',
+        \XmlReader::XML_DECLARATION => 'XML_DECLARATION'
+    );
+
+    protected static $filteredTypes = array(
+        \XmlReader::NONE => true,
+        \XmlReader::ELEMENT => true,
+        \XmlReader::ATTRIBUTE => false,
+        \XmlReader::TEXT => true,
+        \XmlReader::CDATA => true,
+        \XmlReader::ENTITY_REF => false,
+        \XmlReader::ENTITY => false,
+        \XmlReader::PI => false,
+        \XmlReader::COMMENT => true,
+        \XmlReader::DOC => false,
+        \XmlReader::DOC_TYPE => false,
+        \XmlReader::DOC_FRAGMENT => false,
+        \XmlReader::NOTATION => false,
+        \XmlReader::WHITESPACE => false,
+        \XmlReader::SIGNIFICANT_WHITESPACE => false,
+        \XmlReader::END_ELEMENT => false,
+        \XmlReader::END_ENTITY => false,
+        \XmlReader::XML_DECLARATION => true
+    );
+
+    public static function castXmlReader(\XmlReader $reader, array $a, Stub $stub, $isNested)
+    {
+        $dom = $reader->expand();
+
+        $infos = array(
+            'localName' => $reader->localName,
+
+            'depth' => $reader->depth,
+
+            'attributeCount' => $reader->attributeCount,
+            'hasAttributes' => $reader->hasAttributes,
+
+            'hasValue' => $reader->hasValue,
+            'isDefault' => $reader->isDefault,
+            'isEmptyElement' => $reader->isEmptyElement,
+            'nodeType' => new ConstStub(self::$nodeTypes[$reader->nodeType], $reader->nodeType),
+
+            Caster::PREFIX_VIRTUAL.'dom' => new CutStub($dom)
+        );
+
+        if ($reader->hasValue && \XmlReader::TEXT === $reader->nodeType) {
+            $infos['value'] = new CutStub($reader->value);
+        }
+
+        if ($reader->hasAttributes) {
+            $infos[Caster::PREFIX_VIRTUAL.'attributes'] = array();
+
+            foreach ($dom->attributes as $attribute) {
+                $infos[Caster::PREFIX_VIRTUAL.'attributes'][$attribute->name] = $attribute->value;
+            }
+        }
+
+        if (false === static::$filteredTypes[$reader->nodeType]) {
+            $stub->cut += count($infos);
+            return array();
```

Comment:
Added `localName`, `depth` and `nodeType`. I don't think the other are really useful though...